src/Subscription.php

```diff
@@ -671,7 +677,9 @@ public function modifiers()
      */
     public function modifier($id)
     {
-        return $this->modifiers->find($id);
+        return $this->modifiers()->first(function (Modifier $modifier) use ($id) {
+            return $modifier->id() == $id;
```

Comment:
I think we probably want a strict comparison here?

Reply:
Done